### PR TITLE
Fixes #97 - Don't log basic auth credentials in RetryWire

### LINKED ISSUE #97: Basic auth information logged by RetryWire
When RetryWire fails, it logs a warning message that contains the the url, even if that url contains basic auth information used by BasicAuthWire.

I've had a quick look thought about having BasicAuthWire strip off the url credentials after the auth header is created so RetryWire doesn't have to worry about it. However, that would then put a ordering dependency on the Wires which isn't great.

Therefore, just hiding the credentials in the log statement in RetryWire seems a next best.

I'll arrange a pull request

### PR DESCRIPTION
**Pull request by dominicfarr**

### CODE REVIEW

**Comment by dominicfarr** [comment]
Wow - 73 checkstyle violations. 

Not sure I've the will to fix them, which perhaps seems lazy, but frankly your style and my style aren't compatible. 

Please take my pull request for what it is worth. Apply how you see fit.

Regards

**Comment by dmarkov** [comment]
@dominicfarr Many thanks, I will find someone to review it before we merge

**Comment by dmarkov** [comment]
@caarlos0 please review, thanks

**Comment by dominicfarr** [comment]
Sorry about the non-compliance of qulice...far too draconian for me.  But perhaps qulice does provide the special sauce that makes jcabi-http as excellent as it is.

I've highlighted the issue, your team can roll up a fix your way.

**Comment by caarlos0** [comment]
@dominicfarr you should fix the qulice issues to merge the PR, sorry about that :/

File: src/main/java/com/jcabi/http/mock/MkGrizzlyContainer.java [hunk 1]
```diff
@@ -52,7 +53,7 @@
 @SuppressWarnings("PMD.TooManyMethods")
 @EqualsAndHashCode(of = { "adapter", "gws", "port" })
 @Loggable(Loggable.DEBUG)
-public final class MkGrizzlyContainer implements MkContainer {
+public class MkGrizzlyContainer implements MkContainer {
```
Comment (caarlos0) [inline src/main/java/com/jcabi/http/mock/MkGrizzlyContainer.java]: @dominicfarr re-add the `final` please..

**Comment by dominicfarr** [comment]
@caarlos0 as I said, I've no wish to change my style of coding to compile with yours. 

everyone has their own style and i'm i do no intend to get into a silly flame war defending mine, or trying to convince anyone, including yourself, their style is wrong. no one is right, no one is wrong, it is just different. 

please take my pull request as highlight to the error and an example of one possible fix (however poorly styled you may think.)

regards

**Comment by yegor256** [comment]
@dominicfarr maybe this article will help you: http://www.yegor256.com/2014/08/13/strict-code-quality-control.html

**Comment by dominicfarr** [comment]
*deep breath*...oh dear. 

@yegor256 I've read this post, and the comments, and other pull requests, and other blogs postings you have written, and those comments. I feel well read on the whole subject. I've also discussed this with my team of developers. Always edifying when you find subjects like this.

As i've said, your house, your rules, and I'm not the sort of person to try and persuade you otherwise. 

That said, your rules aren't compatible with my rules. 

I still chose to submit a PR in the hope to highlight the issue and offer a solution (however crude and inappropriate.)

What I chose not to do was to make your rules and my rules match; this would have taken a disproportionate time compared to the fix. IMO.

Yet the comments and requests keep coming, which seems, frankly, obtuse.

I really don't mind you rejecting my PR...it was just a friendly nod about this issue.

I've workaround this issue and written my own basic auth wire (with test case). Your version  of BasicAuthWire doesn't have a test case, which I find a tad laughable considering your stance on code quality.

**Comment by dmarkov** [comment]
@caarlos0 **16 mins** added to your account (payment number `AP-9TX34032YY594324J`), many thanks for your contribution! 72 hours and 28 mins spent here.... extra minutes for review comments (c=1)... +16 to your rating, your total score is [+2328](http://www.netbout.com/b/35095?open=rating)

**Pull request closed without merging**